resources/prosody-plugins/mod_auth_token.lua

```diff
@@ -74,18 +74,28 @@ function provider.delete_user(username)
 	return nil;
 end
 
+local function validate_result(session, res, error, reason)
+    if res == false then
+        log("warn",
+            "Error verifying token err:%s, reason:%s", error, reason);
+        session.auth_token = nil;
+        return res, error, reason;
+    end
+end
+
 function provider.get_sasl_handler(session)
 
 	local function get_username_from_token(self, message)
-        local res, error, reason = token_util:process_and_verify_token(session);
 
-        if (res == false) then
-            log("warn",
-                "Error verifying token err:%s, reason:%s", error, reason);
-            session.auth_token = nil;
-            return res, error, reason;
+        -- retrieve custom public key from server and save it on the session
+        local event_result = prosody.events.fire_event("pre-jitsi-authentication-fetch-key", session);
+        if event_result ~= nil then
+            validate_result(session,event_result.res, event_result.error, event_result.reason)
         end
 
+        local res, error, reason = token_util:process_and_verify_token(session);
+        validate_result(session, res, error, reason);
```

Comment:
I believe this function isn't helping us, as we need to check the res value and decide if we are going to return and log an error or not in this top level function, not within validate_result.